src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -6066,23 +6066,15 @@ private URL getResource( final List<String> classPath, final String resource )
                 getLog().error( "MalformedURLException: " + e.getMessage() );
             }
         }
-
-        URLClassLoader javadocClassLoader = new URLClassLoader( urls.toArray( new URL[urls.size()] ), null );
-        try
+        URL resourceURL = null;
+        try ( URLClassLoader javadocClassLoader = new URLClassLoader( urls.toArray( new URL[urls.size()] ), null ) )
         {
-            return javadocClassLoader.getResource( resource );
-        }
-        finally
+            resourceURL = javadocClassLoader.getResource( resource );
+        } catch ( IOException e )

```

Comment:
DANGER: this must be rethrown, not ignored

Reply:
@elharo This IOException is ignored in the original codes, so I don't think we shall rethrow it...

Reply:
The ignored IOException was from closing in the finally block, not from other sources.

Reply:
@elharo no other resource will throw IOException here.
`public URL getResource(String name) {`
IOException is not a RuntimeException. If It can be thrown it must imply it.

Reply:
There's no catch block in the old code. Maybe this isn't thrown at all?

Reply:
@elharo It seems only can throw when closing the `javadocClassLoader`. the original codes igonre it in `finally`. the new codes ignore it in `catch`. same thing IMO.

Reply:
Then try with resources isn't working right here somehow. You shouldn't need to catch IOException that's only thrown on autoclose.

Reply:
> Then try with resources isn't working right here somehow. You shouldn't need to catch IOException that's only thrown on autoclose.
@elharo 
Well, kind of agreed, but that is oracle guys' dicision for forcing us to must catch it...
Or should we revert it back to normal try instead of try-with-resource?

Reply:
There's something I'm not seeing here. Have you tried simply removing the catch block completely? Does that compile?

Reply:
> There's something I'm not seeing here. Have you tried simply removing the catch block completely? Does that compile?

after removing that part:
```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12.622 s
[INFO] Finished at: 2020-06-20T22:46:09+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project maven-javadoc-plugin: Compilation failure
[ERROR] /D:/workspace/maven-javadoc-plugin/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java:[6068,30] unreported exception java.io.IOException; must be caught or declared to be thrown
[ERROR]   exception thrown from implicit call to close() on resource variable 'javadocClassLoader'
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

```

Reply:
Weird. I still don't see why. For now though, just leave the code as it was since autocloseable isn't helping

Reply:
@elharo done.